decidim-assemblies/lib/decidim/assemblies/menu.rb

```diff
@@ -130,7 +130,7 @@ def self.register_admin_assembly_menu!
                         I18n.t("private_users", scope: "decidim.admin.menu.assemblies_submenu"),
                         decidim_admin_assemblies.participatory_space_private_users_path(current_participatory_space),
                         icon_name: "spy-line",
-                        if: allowed_to?(:read, :space_private_user, assembly: current_participatory_space)
+                        if: allowed_to?(:read, :space_private_user, assembly: current_participatory_space) && current_participatory_space.private_space?
```

Comment:
Reading this one more time I realized that we should not only change the menu display but actually the permission, as it's possible to still access to this page even though it's not a private space.

Can you change that please?

Reply:
@andreslucena After spending around 2 hours on the matter, i could not find a easy way to implement the permission, and as a result, i reached the conclusion that the permission system would require a revisit. 

In this particular case, i have tried to remove `space_private_user` from `permission_action.subject` keys in `org_admin_action?`, `process_admin_action?` and `assembly_admin_action?` in order to define it in a stand alone method. 

Doing so, worked fine for processes, but failed for assemblies. I have found that the culprit is the `collaborator_action?` that in the process is checking only if the action is `preview` while in assembly is checking for action `read` and `preview`. As a result, we end up always having allow while reading `space_private_user`.